src/android/toga_android/widgets/detailedlist.py

```diff
@@ -15,7 +17,19 @@ def onClick(self, _view):
             self._impl.interface.on_select(widget=self._impl.interface, row=self._row_number)
 
 
+class OnRefreshListener(android_widgets.SwipeRefreshLayout__OnRefreshListener):
+    def __init__(self, interface):
+        super().__init__()
+        self._interface = interface
+
+    def onRefresh(self):
+        if self._interface.on_refresh:
+            self._interface.on_refresh(self._interface)
+
+
 class DetailedList(Widget):
+    _android_swipe_refresh_layout = None
```

Comment:
Is there a reason this priming is needed?

Reply:
I check the value of this attribute in `_after_on_refresh()`, so I (and my editor's linter) think it's weird to just read an attribute if it might raise AttributeError. This avoids that.

I could check it in `_after_on_refresh()` with `getattr(self, '_android_swipe_refresh_layout')`, but then it's like, do you even know your`self`??

Anyway, that's the idea. Can change it if you want.

I also have to mollify the _actual_ linter apparently, which I'm focusing on.

Reply:
Ok - makes sense. Odd that this hasn't been required anywhere else (that I can think of) in toga...

Reply:
It _is_ odd! It's also the first place I've had to explicitly create a JNI global ref. :) The reason for both of those is that my listener requires an explicit reference to an Android widget to call `setRefreshing(False)`. This is a first for needing an explicit, long-lasting reference to an Android widget.